zuul-core/src/main/java/com/netflix/zuul/origins/NettyOrigin.java

```diff
@@ -77,4 +80,8 @@ RequestAttempt newRequestAttempt(
     IClientConfig getClientConfig();
 
     Registry getSpectatorRegistry();
+
+    default void adjustRetryPolicyIfNeeded(HttpRequestMessage zuulRequest, @Nullable HttpResponse response) {
+        adjustRetryPolicyIfNeeded(zuulRequest);
```

Comment:
Can we avoid this overload by using an `||` or the original method and a new one derived from the response?
That feels cleaner imo.

Reply:
I added a new (awkwardly named) method `originRetryPolicyAdjustmentIfNeeded`

Reply:
Naming is hard. It just needs to convey that the adjustment is inferred by the origin response.